node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/env/DefaultEnvRewriter.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task.util.env;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * Rewrites default-env.txt files.
+ *
+ * @author bjorncs
+ */
+public class DefaultEnvRewriter {
+
+    private final Map<String, Operation> operations = new TreeMap<>();
+    private final Path defaultEnvFile;
+
+    public DefaultEnvRewriter(Path defaultEnvFile) {
+        this.defaultEnvFile = defaultEnvFile;
+    }
+
+    public DefaultEnvRewriter addOverride(String name, String value) {
+        return addOperation("override", name, value);
+    }
+
+    public DefaultEnvRewriter addFallback(String name, String value) {
+        return addOperation("fallback", name, value);
+    }
+
+    public DefaultEnvRewriter addUnset(String name) {
+        return addOperation("unset", name, null);
+    }
+
+    private DefaultEnvRewriter addOperation(String action, String name, String value) {
+        if (operations.containsKey(name)) {
+            throw new IllegalArgumentException(String.format("Operation on variable '%s' already added", name));
+        }
+        operations.put(name, new Operation(action, name, value));
+        return this;
+    }
+
+    public boolean converge() throws IOException {
```

Comment:
There is no need to expose IOException to the caller? Below you can use `uncheck(() -> Files.write())`, or use UnixPath instead of Path.

Reply:
Done